DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,352 @@
+```c++
+struct SingleMMIP {
+  // This kernel has 3 annotated pointers, but since they have no properties
+  // specified, this kernel will result in the same IP component as Example 1.
+  sycl::ext::oneapi::experimental::annotated_arg<int *> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *> z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+```
+
+The following table describes the properties under `sycl::ext::intel::experimental` that you can use to customize how the pointer argument is passed to your component. Only one may be specified at a time. 
+
+| Parameter                 | Description
+|---                        |---
+| `register_map`            | Pass the pointer for this memory-mapped host interface through the IP component's control/status register
+| `conduit`                 | Pass the pointer for this memory-mapped host interface through a conduit interface 
+
+
+You can use the following parameters found under `sycl::ext::intel::experimental` and `sycl::ext::oneapi::experimental` to configure your IP component's Avalon memory-mapped host interfaces:
+
+| Parameter                | Default Value | Description
+|---                       |---            |---
+|  `buffer_location<id>`   | N/A           | The address space of the interface that associates with the host. Each unique buffer location will result in a unique Avalon memory-mapped host interface. When `buffer_location` is not specified, then corresponding LSUs will connect to all of the IP's Avalon memory-mapped host interfaces.
+| `awidth<width>`          | 41            | Width of the Avalon memory-mapped host address bus in bits. 
+| `dwidth<width>`          | 64            | Width of the Avalon memory-mapped host data bus in bits. 
+| `latency<value>`         | 1             | Read/Write latency for downstream fixed-latency Avalon memory-mapped agent. For a variable-latency Avalon memory-mapped agent, set `latency<0>`. 
+| `read_write_mode<mode>`  | `read_write`  | Port direction of the interface. (`read_write`, `read` or `write`) 
+| `maxburst<value>`        | 1             | Maximum number of data transfers that can associate with a read or write request. 
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+| `stable`                 | N/A | User guarantee that the pointer will not change between pipelined invocations of the kernel. The compiler uses this to further optimize the kernel.
+
+These parameters can be used to improve the performance of `Example 1` by ensuring that each pointer points to data in a dedicated Avalon memory-mapped agent memory, like this:
+
+![](assets/dedicated_avhost.svg)
+
+#### Example 3: A kernel with dedicated Avalon memory-mapped host interfaces
+```c++
+constexpr int kBL1 = 0;
```

Comment:
I think that buffer location 0 is prohibited in 2024.0. @tiwaria1 can you confirm if my understanding is correct?

Reply:
We technically relaxed that limitation but due to some new late discovered issues effectively yes. Hakim, please make these be `1,2,3` instead of `0,1,2`

> Should they all be defined as constexpr beforehand

We defined buffer locations before hand since they must be used again on the malloc* call. That is not true for the other properties so we do not need to. We do need to specify alignment twice so we could do that with alignment.

Reply:
Hakim, let's talk offline about specifying alignment to malloc* calls